### ISSUE TITLE
Solderer dying

### ISSUE THREAD

**Issue by raoulvdberge** [raoulvdberge/refinedstorage#104]

**Comment by RealGrep** [comment]
I should note a Twitch viewer commented that he had to break and replace various other units and cables after leaving and coming back, so it's probably not restricted to just the solderer. Did he bother filing a bug report for it rather than just complaining about it in a Twitch channel? Don't be silly. XD

**Comment by raoulvdberge** [comment]
@RealGrep Can you try this debug build in a fresh world with just a solderer? Let me know what the console says.
[refinedstorage-debug.zip](https://github.com/raoulvdberge/refinedstorage/files/310839/refinedstorage-debug.zip)

**Comment by RealGrep** [comment]
Can't seem to get it to freeze up in a fresh world with only the solderer, at the moment. Doing the same thing as usual and going to the nether, waiting a couple of minutes, returning and hitting the redstone control mode button. Do you still want the logs? Shall I try it in my normal world where it locks up?

**Comment by raoulvdberge** [comment]
Yes, please do.

**Comment by RealGrep** [comment]
Spoke too soon. Managed. I wonder if it has anything to do with TheOneProbe. It's spewing errors there. If not, at least the NullPointerException might be a clue. Anyway, I zipped up the entire log. To find the relevant section of the log, search for a few asterisks, i.e. "***".

https://www.dropbox.com/s/ltpe188zevnlj51/fml-server-latest-rs1.zip?dl=0

So basically, I moved it away from the spawn chunk to make sure it wasn't loaded when I left. I put the nether portal slightly away from the solderer (one chunk away diagonally). Went to the nether, waited a couple of minutes, and returned. The redstone mode button no longer worked. Solderer seems to stop ticking.

**Comment by RealGrep** [comment]
There's a possibility the TOP NPE is from this: https://github.com/McJty/TheOneProbe/issues/14 . And if so, that might fix it. Though the solderer isn't really multi-part, so not sure this applies.

**Comment by raoulvdberge** [comment]
Huh:
```
[14:22:01] [Server thread/INFO] [STDOUT/]: [refinedstorage.tile.TileMachine:func_73660_a:54]: [REFINED STORAGE DEBUG] Machine Tick 1280 (Connected=false,controller=null,pos=BlockPos{x=517, y=64, z=-210})
[14:22:02] [Server thread/DEBUG] [FML/]: The world 1a53be82 (Addle's Realm) may have leaked: seen 140 times.
**************** 14:22 go to nether, solderer so far ok
```
Your solderer wasn't even connected when you went to the nether?

**Comment by RealGrep** [comment]
Correct. You did say just the solderer. :)

**Comment by raoulvdberge** [comment]
lol, that is true... I actually need the solderer to be connected to a controller (which is not technically a machine) before you go the the nether :)

**Comment by RealGrep** [comment]
Ok, will do so.

**Comment by raoulvdberge** [comment]
About the TOP error, the solderer *does* check for null:
Solderer:
```
if (capability == CapabilityItemHandler.ITEM_HANDLER_CAPABILITY && facing != null) {
```
TOP:
```
        if (te != null && te.hasCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, null)) {
            IItemHandler capability = te.getCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, null);
```

**Comment by raoulvdberge** [comment]
This should fix The One Probe error:

[refinedstorage-debug.zip](https://github.com/raoulvdberge/refinedstorage/files/310958/refinedstorage-debug.zip)

**Comment by RealGrep** [comment]
This is the log with the solderer connected to a creative controller:

https://www.dropbox.com/s/addzyr6l6wmqeq9/fml-server-latest-rs2.zip?dl=0

**Comment by raoulvdberge** [comment]
What happened here?

```
[13:58:23] [Server thread/INFO] [STDOUT/]: [refinedstorage.tile.TileMachine:func_73660_a:54]: [REFINED STORAGE DEBUG] Machine Tick 1500 (Connected=true,controller=refinedstorage.tile.controller.TileController@7a5f83ce,pos=BlockPos{x=126, y=64, z=293})
[13:58:24] [Server thread/INFO] [STDOUT/]: [refinedstorage.tile.TileMachine:func_73660_a:66]: [REFINED STORAGE DEBUG] BlockPos{x=126, y=64, z=293} My state changed to false
[13:58:24] [Server thread/INFO] [STDOUT/]: [refinedstorage.tile.TileMachine:func_73660_a:54]: [REFINED STORAGE DEBUG] Machine Tick 1520 (Connected=false,controller=null,pos=BlockPos{x=126, y=64, z=293})
```
Did you change the redstone control of the controller?

Also, check if the error still persists with the other debug version I posted.

**Comment by RealGrep** [comment]
Yep, before I go to the nether, I click it to make sure it's ok. I click it again when I return to see if it's frozen up or not.

About to test with the new version.

**Comment by RealGrep** [comment]
Ok, same basic result, except the errors from TOP are gone (so as I figured was likely, they are unrelated, but still nice to have them gone):

https://www.dropbox.com/s/1c7xupl88pw9n4a/fml-server-latest-rs3.zip?dl=0

**Comment by raoulvdberge** [comment]
Did the machine stop ticking or you just left the game quickly?

**Comment by RealGrep** [comment]
Still in the game next to it, and I see no ticking messages. The last line on the console is still: "[16:18:11] [Server thread/INFO] [FML]: Unloading dimension -1"

**Comment by raoulvdberge** [comment]
Okay, interesting...

**Comment by RealGrep** [comment]
Interestingly, the redstone control button seems to work ok on the creative controller. No messages are generated in the console when I click on it, but it updates and changes modes.

**Comment by raoulvdberge** [comment]
Does the redstone button work on other machines like the grid or importer after coming back from the nether?

**Comment by RealGrep** [comment]
I'll check that now.

**Comment by RealGrep** [comment]
Nope, I put down a ton of machines. None of the buttons work in any of them, except the creative controller.

https://www.dropbox.com/s/xz2txil4olqe4xm/fml-server-latest-rs4.zip?dl=0

What I find interesting is it seems things are ok until the Nether is actually unloaded, at which point all the messages from RS stop.

**Comment by RealGrep** [comment]
Yep, the two are related somehow. What I did was put down a chunkloader in the nether to keep it loaded (The ExtraUtils2 one, along with a bunch of lava generators to power it).

Did the same test as usual and waited a couple of minutes in the nether. Came back, and all the machines seem to be responsive, and I can change the state of all the buttons.

That's rather weird!

**Comment by raoulvdberge** [comment]
So if you keep the nether loaded while the machines are in the overworld, everything keeps working? O_O

**Comment by RealGrep** [comment]
Yep! Sort of weird, huh? lol

**Comment by RealGrep** [comment]
Odd, I just repeated it, the nether didn't unload, but the buttons are unresponsive. Might be a red herring.

**Comment by raoulvdberge** [comment]
Ok, I removed the check if redstone is enabled for testing. Can you retest with this?
[refinedstorage-debug.zip](https://github.com/raoulvdberge/refinedstorage/files/310999/refinedstorage-debug.zip)

**Comment by RealGrep** [comment]
Same basic test sequence with new debug build: https://www.dropbox.com/s/ypvusruh0p2d1rd/fml-server-latest-rs5.zip?dl=0

**Comment by raoulvdberge** [comment]
Hmm,  can you test this build?

Really, thank you for the testing.

[refinedstorage-debug130616.zip](https://github.com/raoulvdberge/refinedstorage/files/311012/refinedstorage-debug130616.zip)

**Comment by RealGrep** [comment]
Here you go: https://www.dropbox.com/s/wh34e26rlh1j64e/fml-server-latest-rs6.zip?dl=0

And you're welcome, I'm glad to help out. :)

**Comment by raoulvdberge** [comment]
Okay, can you now try with this build:

[refinedstorage-debug.zip](https://github.com/raoulvdberge/refinedstorage/files/318930/refinedstorage-debug.zip)

**Comment by RealGrep** [comment]
Sadly, bug still exists. First test worked ok, so I stayed a bit longer in the nether before returning. This time, the redstone mode buttons are unresponsive again. Sorry. :(

**Comment by raoulvdberge** [comment]
Really weird issue. Will definitely look at it 100% at the weekend, considering as this is the only remaining bug open.

**Comment by RealGrep** [comment]
It is. Seems to be better if I keep my area chunk loaded, so definitely seems to be related to chunk loading/unloading.

And yeah, pretty much the only real bug I know of left in it. heh

**Comment by raoulvdberge** [comment]
Please send a zip with the pack and configs to raoulvdberge at gmail dot com

**Comment by RealGrep** [comment]
Done :)

**Comment by WhiteFire-Sondergaard** [comment]
I am in my world seeing a more generalized version of this problem. Sometimes if I am off roaming the world or in another dimension I come back to my base and the controller does not see _any_ of it's connected devices. At first I thought it was a power issue, but after putting a huge capacitor to make sure it wasn't fluctuation in my power grid... nope.

It's easily fixed by going to the controller and toggling the redstone mode so it goes off then on, where it rediscovers all it's devices. However this is bad news for unattended automation.

All chunks involved are chunk loaded with Ender Anchors (random things) in an attempt to resolve this. Didn't help.

**Comment by WhiteFire-Sondergaard** [comment]
Possibly (probably?) related: the controller is in a different chunk than all it's devices other than it's power source. All should be chunk loaded.

**Comment by raoulvdberge** [comment]
I know what the cause is of this issue. I could do a simple hack to fix it, but I'm reworking the machine system once more, so it'll be fixed as soon as the rework is done.

**Comment by RealGrep** [comment]
That's good news! Congrats on finding it. :)

Not a problem, best to fix it properly, IMO. In the meanwhile, keeping it chunk loaded seems to help.

**Comment by raoulvdberge** [comment]
Should be fixed now! @RealGrep, please hang in there. Will post a test build for you when the dev branch is a bit more stable.

**Comment by RealGrep** [comment]
Well, did the usual test sequence three times, and I can't replicate the bug any more. So looks like it's fixed, unless I was just unlucky in my testing. Good job! :D

**Comment by raoulvdberge** [comment]
![lol](http://gifsec.com/wp-content/uploads/GIF/2014/05/Party-reply-gif.gif?gs=a)

**Comment by raoulvdberge** [comment]
If you DO see it again though, please do not hesitate to make me unhappy again xD

**Comment by RealGrep** [comment]
Will do! XD
